aws_lambda_powertools/utilities/data_classes/dynamo_db_stream_event.py

```diff
@@ -176,28 +85,43 @@ class StreamViewType(Enum):
 
 
 class StreamRecord(DictWrapper):
+    def __init__(self, data: Dict[str, Any]):
+        """StreamRecord constructor
+
+        Parameters
+        ----------
+        data: Dict[str, Any]
+            Represents the dynamodb dict inside DynamoDBStreamEvent's records
+        """
+        super().__init__(data)
+        self._deserializer = TypeDeserializer()
```

Comment:
```suggestion
class StreamRecord(DictWrapper):
    _deserializer = TypeDeserializer()  # init once instead of per record

    def __init__(self, data: Dict[str, Any]):
        """StreamRecord constructor

        Parameters
        ----------
        data: Dict[str, Any]
            Represents the dynamodb dict inside DynamoDBStreamEvent's records
        """
        super().__init__(data)
```